skimage/feature/tests/test_sift.py

```diff
@@ -129,3 +129,35 @@ def test_no_descriptors_extracted_sift():
     detector_extractor = SIFT()
     with pytest.raises(RuntimeError):
         detector_extractor.detect_and_extract(img)
+
+
+def test_features_dtype():
+    detector_extractor = SIFT()
+    positions_oct = np.array([[1.0, 1.0]])
+    sigmas_oct = np.array([0.8])
+    octaves = np.array([0])
+    scales_oct = np.array([1])
+    gaussian_scalespace = detector_extractor._create_scalespace(img)
+    detector_extractor._compute_orientation(
+        positions_oct, scales_oct, sigmas_oct, octaves, gaussian_scalespace
+    )
+    assert (
+        detector_extractor.positions is not None
+        and detector_extractor.positions.dtype == positions_oct.dtype
+    )
```

Comment:
Could we make these asserts more explicit by just directly comparing to the desired dtype? And I'm not sure why the `is None` check is needed? It seems to me that the the values should not fluctuate in this test.

```suggestion
    assert (
        detector_extractor.positions is not None
        and detector_extractor.positions.dtype == positions_oct.dtype
    )
```